internal/util/validation/formats.go

```diff
@@ -0,0 +1,54 @@
+package validation
+
+import (
+	"regexp"
+)
+
+const (
+	dns1123LabelFmt       string = `[a-z0-9]([-a-z0-9]*[a-z0-9])?`
+	dns1123LabelMaxLength int    = 63
+)
+
+var GenericNameRegexp = regexp.MustCompile("^" + dns1123LabelFmt + "$")
+
+func ValidateGenericName(name *string, path string) []error {
+	return ValidateString(name, path, 1, dns1123LabelMaxLength, GenericNameRegexp, dns1123LabelFmt)
+}
+
+const (
+	// as per https://github.com/containers/image/blob/main/docker/reference/regexp.go
+	ociDomainCompFmt           string = `(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])`
+	ociNameCompFmt             string = `[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?`
+	OciImageDomainFmt          string = ociDomainCompFmt + `(?:[.]` + ociDomainCompFmt + `)*` + `(?::[0-9]+)?`
+	OciImageNameFmt            string = `(?:` + OciImageDomainFmt + `\/)?` + ociNameCompFmt + `(?:\/` + ociNameCompFmt + `)*`
+	OciImageTagFmt             string = `[\w][\w.-]{0,127}`
+	OciImageDigestFmt          string = `[A-Za-z][A-Za-z0-9]*(?:[-_+.][A-Za-z][A-Za-z0-9]*)*[:][[:xdigit:]]{32,}`
+	OciImageReferenceFmt       string = `(` + OciImageNameFmt + `)(?:\:(` + OciImageTagFmt + `))?(?:\@(` + OciImageDigestFmt + `))?`
+	OciImageReferenceMaxLength int    = 2048
+)
+
+// capture(namePat)
+// optional(literal(":"), capture(tag))
+// optional(literal("@"), capture(digestPat))
+
+var (
+	OciImageReferenceRegexp = regexp.MustCompile("^" + OciImageReferenceFmt + "$")
+)
+
+// Validates an OCI image reference.
+func ValidateOciImageReference(s *string, path string) []error {
+	return ValidateString(s, path, 1, OciImageReferenceMaxLength, OciImageReferenceRegexp, OciImageReferenceFmt, "quay.io/flightctl/flightctl:latest")
+}
+
+const (
+	// as per https://docs.github.com/en/get-started/using-git/dealing-with-special-characters-in-branch-and-tag-names#naming-branches-and-tags
+	GitRevisionFmt string = `[a-zA-Z]([a-zA-Z0-9\.\-\_\/])*`
```

Comment:
This regexp does not work well with hashes, eg

```
spec.config[].gitRef.targetRevision: Invalid value: "9fac431b7f4f319ead0195034064012e732bbb0c": Invalidpattern (regex used for validation is '[a-zA-Z]([a-zA-Z0-9\.\-\_\/])*')
```

So a PUT with a device having such a hash would fail.

Reply:
Done.

Reply:
Worked well now :)